android/modules/ui/src/java/ti/modules/titanium/ui/widget/picker/TiUINativePicker.java

```diff
@@ -237,39 +237,45 @@ protected void refreshNativeView()
 			if (rememberSelectedRow >= 0) {
 				selectRow(0, rememberSelectedRow, false);
 			}
-
+			// The new adapter has been set.
+			// If the Spinner has been drawn reattach the onItemSelected listener here.
+			// If it has not been drawn yet, the listener will be attached in the
+			// onLayout lifecycle event.
+			if (nativeViewDrawn) {
+				spinner.setOnItemSelectedListener(onItemSelectedListener);
+			}
 		} catch (Throwable t) {
 			Log.e(TAG, "Unable to refresh native spinner control: " + t.getMessage(), t);
-		} finally {
-			suppressChangeEvent = false;
-			spinner.setOnItemSelectedListener(this);
 		}
 	}
 
-	@Override
-	public void onItemSelected(AdapterView<?> parent, View view, int position, long itemId)
-	{
-		if (!firstSelectedFired) {
+	final OnItemSelectedListener onItemSelectedListener = new OnItemSelectedListener() {
+		@Override
+		public void onItemSelected(AdapterView<?> parent, View view, int position, long itemId)
+		{
+			/*if (!firstSelectedFired) {
 			// swallow the first selected event that gets fired after the adapter gets set, so as to avoid
 			// firing our change event in that case.
 			firstSelectedFired = true;
 			return;
-		}
-		fireSelectionChange(0, position);
+		}*/
```

Comment:
Since you've removed the `firstSelectedFired` member variable, we might as well as delete this commented out code block too.